### DISCUSSION TITLE
sessions view in oracle is wrong and different between oracle sql developer

### DISCUSSION THREAD

**Discussion by leezm** [dbeaver/dbeaver#15944] [General]
#### System information: 
- win11
- DBeaver EE 22.0.0.202203131528
- Additional extensions

#### Connection specification:
- Oracle 11g
- ojdbc8
- NO

#### Describe the problem you're observing:
When I view oracle sessions in dbeaver,I found out it was wrong.It's different between oracle sql developer.
#### Steps to reproduce, if exist:
Double click sessions in oracle db view.
#### Include any warning/errors/backtraces from the logs

Dbeaver use sql:
SELECT s.*, sq.SQL_FULLTEXT, io.*
FROM GV$SESSION s, gv$sql sq, gv$sess_io io
WHERE s.sql_address = sq.address(+)
 AND s.sql_hash_value = sq.hash_value(+) AND s.sid = io.sid(+) AND s.inst_id = io.inst_id(+) AND s.sql_child_number = sq.child_number (+) AND s.TYPE = 'USER' AND s.STATUS <> 'INACTIVE'

Oracle sql developer use sql:

                                                  with vs as (select rownum rnum,
                                                  							  inst_id,
                                                                              sid,
                                                                              serial#,
                                                                              status,
                                                                              username,
                                                                              last_call_et,
                                                                              command,
                                                                              machine,
                                                                              osuser,
                                                                              module,
                                                                              action,
                                                                              resource_consumer_group,
                                                                              client_info,
                                                                              client_identifier,
                                                                              type,
                                                                              terminal,
                                                                              sql_id,
                                                                              sql_child_number
                                                                         from gv$session) 
                                                             select vs.inst_id, vs.sid ,serial# serial, vs.sql_id, vs.sql_child_number,
                                                                    vs.username "Username",
                                                                    case when vs.status = 'ACTIVE' 
                                                                              then last_call_et 
                                                                         else null end "Seconds in Wait",
                                                                    (select command_name from v$sqlcommand where command_type = vs.command ) "Command",
                                                                    vs.machine "Machine",
                                                                    vs.osuser "OS User", 
                                                                    lower(vs.status) "Status",
                                                                    vs.module "Module",
                                                                    vs.action "Action",
                                                                    vs.resource_consumer_group,
                                                                    vs.client_info,
                                                                    vs.client_identifier
                                                               from vs 
                                                              where vs.USERNAME is not null
                                                                and nvl(vs.osuser,'x') <> 'SYSTEM'
                                                                and vs.type <> 'BACKGROUND'
                                                                order by 1,2,3
                                               
Check out it PLZ!!

**Comment by p91paul** [comment]
having extensively used the feature for years, "wrong" is not the adjective I'd use. I think you'll have to be more specific: sure the queries are different, but you should point out something that is correct on sql developer and wrong (not just different) on dbeaver.

**Comment by leezm** [comment]
> having extensively used the feature for years, "wrong" is not the adjective I'd use. I think you'll have to be more specific: sure the queries are different, but you should point out something that is correct on sql developer and wrong (not just different) on dbeaver.

Yes,you're right.But I remember that the previous version showed it right.I found that Dbeaver now has significantly fewer sessions，so I say it's wrong.

**Comment by p91paul** [comment]
there are a couple buttons at the bottom to show background and inactive sessions. Background are excluded in sql dev according to your query, but maybe showing inactive does what you need?

**Comment by leezm** [comment]
> there are a couple buttons at the bottom to show background and inactive sessions. Background are excluded in sql dev according to your query, but maybe showing inactive does what you need?

On,yes!show inactive  button looks like right!But the previous version showed all directly?

**Comment by LonwoLonwo** [comment]
The previous version didn't show all directly.
